Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using AggieEnterpriseApi;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+using Sloth.Core.Models;
+using Sloth.Core.Resources;
+using Sloth.Core.Services;
+
+namespace Sloth.Core.Jobs
+{
+    public class AggieEnterpriseJournalJob
+    {
+        private readonly SlothDbContext _context;
+        private readonly IAggieEnterpriseService _aggieEnterpriseService;
+
+        public static string JobName = "AggieEnterprise.JournalProcessor";
+
+        public AggieEnterpriseJournalJob(SlothDbContext context, IAggieEnterpriseService aggieEnterpriseService)
+        {
+            _context = context;
+            _aggieEnterpriseService = aggieEnterpriseService;
+        }
+
+        public async Task UploadTransactions(ILogger log)
+        {
+            try
+            {
+                // fetch staged transactions with FinancialSegmentString populated
+                var transactions = await _context.Transactions
+                    .Where(t => t.Transfers.Any(tran => tran.FinancialSegmentString != null))
+                    .Where(t => t.Status == TransactionStatuses.Scheduled)
+                    .Include(t => t.Transfers)
+                    .Include(t => t.Source)
+                    .ThenInclude(s => s.Team)
+                    .Include(t => t.ReversalOfTransaction)
+                    .ToListAsync();
+
+                if (!transactions.Any())
+                {
+                    log.Information("No scheduled transactions found");
+                    return;
+                }
+
+                // group transactions by origin code and feed
+                var groups = transactions.GroupBy(t => t.Source);
+                foreach (var group in groups)
+                {
+                    try
+                    {
```

Comment:
This try/catch block seems redundant, given the nested loop has its own try/catch block.